### PR TITLE
fix: api_url setter methods & chunked upload

### PR DESCRIPTION
**Pull request by pawanpaudel93**
# Issues fixed

- api_url setter was throwing an error as it contained set_ prefix on the setter method.

- Also, in the chunked transaction uploading process it was having issues as it was trying to encode a value that is bytes already and trying to decode a value that is a string already.

### CODE REVIEW

**Comment by pawanpaudel93** [comment]
@xloem Please check this PR.

**Comment by xloem** [comment]
thanks for this. I don’t really use transaction.py, it’s left over from before the fork. Are you able to comment on what experience or observation you had that stimulated the changes, so I can understand them more easily?

**Comment by pawanpaudel93** [comment]
@xloem api_url setter was throwing an error as it contained set_ prefix on the setter method. Also, in the chunked transaction uploading process it was having issues as it was trying to encode a value that is bytes already and trying to decode a value that is a string already. You can also check on what was changed on the commits.

**Comment by pawanpaudel93** [comment]
@xloem making these changes worked to fix the above issues. So can you please try the api_url setter and the chunked upload on your main branch and see if it's having errors and is fixed by my PR?

**Comment by xloem** [comment]
I'm sorry, I don't understand, and you're explanation doesn't mention the 2 different tag formats nor describe the change in the setter naming convention you made. I would love it if you would fork the repository and take over maintenance of it. I have been losing my mind and it is hard for me to understand things.

File: ar/transaction.py [hunk 1]
```diff
@@ -170,8 +170,9 @@ def tobytes(self):
     @property
     def api_url(self):
         return self.peer.api_url
+    
     @api_url.setter
-    def set_api_url(self, api_url):
+    def api_url(self, api_url):
```
Comment (xloem) [inline ar/transaction.py:175]: pawan paudel is saying that the set_ prefix here is wrong, and throws an error. There is a working function at https://github.com/xloem/pyarweave/blob/d7d41ec35c05cdc403cfbd021820d169bd04068a/ar/bundle.py#L122 that uses the same naming convention as pawan paudel's, so this is likely a needed change and should be merged.

**Comment by xloem** [comment]
I figured you are using transaction.py more than me so you are probably more right about changes that should happen to it!
Sorry I got so tense around the pull request. Thank you so much for fixing things !!!!

**Comment by pawanpaudel93** [comment]
@xloem Thank you for the merge. It is a great package with a lot of additions to arweave-python-client. Also, can you please make a new version release to PyPI?

**Comment by xloem** [comment]
I made a new version but ran into local system issues releasing to pypi. Are you able to do it if I add you to the pypi account?

**Comment by pawanpaudel93** [comment]
@xloem New Version is released.

**Pull request merged**